firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/ApkUpdater.java

```diff
@@ -110,26 +112,33 @@ UpdateTaskImpl updateApk(
   }
 
   private void installApk(File file, boolean showDownloadNotificationManager) {
-    apkInstaller
-        .installApk(file.getPath())
-        .addOnSuccessListener(
+    lifeCycleNotifier
+        .getForegroundActivity()
+        .onSuccessTask(
             taskExecutor,
-            unused -> {
-              synchronized (updateTaskLock) {
-                safeSetTaskResult(cachedUpdateTask);
-              }
-            })
-        .addOnFailureListener(
-            taskExecutor,
-            e -> {
-              postUpdateProgress(
-                  file.length(),
-                  file.length(),
-                  UpdateStatus.INSTALL_FAILED,
-                  showDownloadNotificationManager);
-              setUpdateTaskCompletionErrorWithDefault(
-                  e, ErrorMessages.APK_INSTALLATION_FAILED, Status.INSTALLATION_FAILURE);
-            });
+            activity ->
+                apkInstaller
+                    .installApk(file.getPath(), activity)
+                    .addOnSuccessListener(
```

Comment:
Can we chain this and `addOnFailureListener` below off of `onSuccessTask` instead of `installApk`? It should not make much of a difference except that if the Task returned by getForegroundActivity() did fail for some reason, then we would correctly fail this class's Task. Right now I believe that Task would hang in that case. Maybe we should add a test for that case, even though it's unlikely.